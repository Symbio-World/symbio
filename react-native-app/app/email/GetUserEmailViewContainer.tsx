import * as React from 'react'
import { GetUserEmailView } from './GetUserEmailView'
import { useAuth } from '../auth'
import { saveEmail } from './saveEmail'

type Props = {
  onSave?: () => void
  onClose?: () => void
}
export const GetUserEmailViewContainer: React.FC<Props> = ({
  onSave = () => {},
  onClose = () => {},
}) => {
  const { user, setEmail } = useAuth()

  const handleSubmit = (email: string) => {
    if (user) {
      saveEmail(user.id, email)
      setEmail(email)
      onSave()
    }
  }

  return <GetUserEmailView onSubmit={handleSubmit} onClose={onClose} />
}
