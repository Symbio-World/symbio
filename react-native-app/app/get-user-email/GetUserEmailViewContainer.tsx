import * as React from 'react'
import { GetUserEmailView } from './GetUserEmailView'
import { useAuth } from '../auth'
import { saveEmail } from './saveEmail'

type Props = {
  onSave?: () => void
}
export const GetUserEmailViewContainer: React.FC<Props> = ({
  onSave = () => {},
}) => {
  const { user } = useAuth()

  const handleSubmit = (email: string) => {
    if (!user) {
      throw new Error('This should never happen, as user is in the context')
    }
    saveEmail(user.id, email)
    onSave()
  }

  return <GetUserEmailView onSubmit={handleSubmit} />
}
