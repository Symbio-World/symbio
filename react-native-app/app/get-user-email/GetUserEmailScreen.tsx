import * as React from 'react'
import { RouteProp } from '@react-navigation/native'
import { StackNavigationProp } from '@react-navigation/stack'
import { GetUserEmailViewContainer } from './GetUserEmailViewContainer'
import { ModalStackParamList } from '../navigation/Navigation'

type Props = {
  navigation: StackNavigationProp<ModalStackParamList, 'GetUserEmailScreen'>
  route: RouteProp<ModalStackParamList, 'GetUserEmailScreen'>
}
export const GetUserEmailScreen: React.FC<Props> = () => {
  return <GetUserEmailViewContainer />
}
