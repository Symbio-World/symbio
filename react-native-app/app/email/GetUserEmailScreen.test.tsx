import * as React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import { GetUserEmailScreen } from './GetUserEmailScreen'
import { GetUserEmailViewContainer } from './GetUserEmailViewContainer'

describe('GetUserEmailScreen', () => {
  let navigation: any = {}
  const route: any = {
    params: {},
  }

  it('renders correctly', () => {
    const { toJSON } = render(
      <GetUserEmailScreen route={route} navigation={navigation} />,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('navigates out on save', () => {
    navigation = { pop: jest.fn() }
    const { getByType } = render(
      <GetUserEmailScreen route={route} navigation={navigation} />,
    )
    fireEvent(getByType(GetUserEmailViewContainer), 'onSave')
    expect(navigation.pop).toHaveBeenCalledTimes(1)
  })

  it('navigates out on close', () => {
    navigation = { pop: jest.fn() }
    const { getByType } = render(
      <GetUserEmailScreen route={route} navigation={navigation} />,
    )
    fireEvent(getByType(GetUserEmailViewContainer), 'onClose')
    expect(navigation.pop).toHaveBeenCalledTimes(1)
  })
})
