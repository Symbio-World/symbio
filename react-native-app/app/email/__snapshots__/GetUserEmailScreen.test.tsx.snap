// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetUserEmailScreen renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "marginTop": 24,
      },
      Object {
        "marginBottom": 12,
      },
      Object {
        "marginHorizontal": 20,
      },
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 24,
          },
          Object {
            "marginBottom": 24,
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      Help us improve!
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
          },
          Object {
            "marginBottom": 16,
          },
        ]
      }
    >
      Hi there, thanks for using Symbio! We want to make it better, so weâ€™re seeking your help to find out what you need. Will you be willing to take part in a short customer interview? Let us know by entering your email below so we can contact you. ðŸ™‚
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
          },
          Object {
            "marginBottom": 12,
          },
        ]
      }
    >
      Your email
    </Text>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoFocus={true}
        keyboardType="email-address"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        rejectResponderTermination={true}
        returnKeyType="send"
        style={
          Array [
            Object {
              "borderWidth": 1,
            },
            Object {
              "borderColor": "#a0aec0",
            },
            Object {
              "height": 48,
            },
            Object {
              "borderRadius": 8,
            },
            Object {
              "padding": 12,
            },
            Object {
              "color": "#000",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "space-evenly",
          },
          Object {
            "position": "absolute",
          },
          Object {
            "bottom": 0,
          },
          Object {
            "width": "100%",
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "padding": 8,
              },
              Object {
                "backgroundColor": "#fff",
              },
              Object {
                "borderRadius": 8,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                },
                Object {
                  "borderWidth": 2,
                },
                Object {
                  "borderColor": "#B35340",
                },
                Object {
                  "alignItems": "center",
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "width": 48,
                },
                Object {
                  "height": 48,
                },
                Object {
                  "borderRadius": 9999,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "width": 20,
                  },
                  Object {
                    "height": 20,
                  },
                ]
              }
            >
              <SvgMock
                height="100%"
                width="100%"
              />
            </View>
          </View>
          <View>
            <Text
              style={
                Array [
                  Object {
                    "textAlign": "center",
                  },
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#B35340",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              close
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "padding": 8,
              },
              Object {
                "backgroundColor": "#fff",
              },
              Object {
                "borderRadius": 8,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#B35340",
                },
                Object {
                  "alignItems": "center",
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "width": 48,
                },
                Object {
                  "height": 48,
                },
                Object {
                  "borderRadius": 9999,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "width": 20,
                  },
                  Object {
                    "height": 20,
                  },
                ]
              }
            >
              <SvgMock
                height="100%"
                width="100%"
              />
            </View>
          </View>
          <View>
            <Text
              style={
                Array [
                  Object {
                    "textAlign": "center",
                  },
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#B35340",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              send
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
